### PR TITLE
feat: Configure servers to accept inputs via CLI args rather than env vars

### PR DESCRIPTION
**Pull request by jsantell**
Enables consistent parallel testing.

* This is mostly for `BUILDER_ADDRESS`, though added `clap` to builder for consistency with runtime
* `clap` could be removed for manual parsing in the name of minimizing dependencies
* As of now, the builder/runtime Dockerfiles are swapped with each other -- once using these images in a pipeline, will need to be updated

### CODE REVIEW

**Comment by cdata** [comment]
Can you provide an explanation of the inconsistency you encountered?

**Comment by jsantell** [comment]
> Can you provide an explanation of the inconsistency you encountered?

Mostly module not found errors during tests. Typically not using `nextest`

**Comment by cdata** [comment]
Environment variable-based configuration is very close to the practical mechanism for configuration in our only use case (cloud deployment). `clap` parsing is nice, but AFAIK it's pretty close in practice to `from_str` behavior.  And, CLI-based configuration will become a moment of indirection.

Keep in mind what deployment entails: k8s configuration passes to environment variables before it makes it to the container image.

Anyway, I'm not opposed to adding in the CLI, but if the only thing we're addressing is consistency across `builder` and `runtime`, I don't think adding the CLI is necessary.

**Comment by jsantell** [comment]
> Environment variable-based configuration is very close to the practical mechanism for configuration in our only use case (cloud deployment). `clap` parsing is nice, but AFAIK it's pretty close in practice to `from_str` behavior. And, CLI-based configuration will become a moment of indirection.
> 
> Keep in mind what deployment entails: k8s configuration passes to environment variables before it makes it to the container image.

Advantage for args is testing and local development, any deployment mechanism is mostly indifferent to args vs env vars. 

> Anyway, I'm not opposed to adding in the CLI, but if the only thing we're addressing is consistency across `builder` and `runtime`, I don't think adding the CLI is necessary.

Consistency w/r/t the builder using the same pattern as the runtime, both introduced here. I don't feel strongly w/r/t clap vs manual parsing, just want to avoid concurrently using a global namespace in tests

File: rust/common-runtime/src/bin/runtime.rs [hunk 1]
```diff
@@ -10,26 +10,41 @@ pub fn main() {
...
+    #[derive(clap::Parser)]
+    #[command(version, about, long_about = None)]
+    struct Cli {
+        /// Set runtime server to listen on provided port.
+        #[arg(short, long, default_value_t = 8081)]
+        port: u16,
+
+        /// URL to the build server.
+        #[arg(short, long)]
+        builder_address: Option<String>,
```
Comment (cdata) [inline rust/common-runtime/src/bin/runtime.rs:27]: Can we make this a `Uri` instead of `String`?

Comment (jsantell) [reply rust/common-runtime/src/bin/runtime.rs]: Indeed that works, must be `url::Url` that doesn't like being parsed here IIRC, updating

**Review by cdata — APPROVED** [review]
Woot!

**Pull request merged**